compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1334,7 +1334,8 @@ private void defineErrorDetails(List<BLangTypeDefinition> typeDefNodes, SymbolEn
     }
 
     private void defineFields(List<BLangTypeDefinition> typeDefNodes, SymbolEnv pkgEnv) {
-        for (BLangTypeDefinition typeDef : typeDefNodes) {
+        for (int i = 0; i < typeDefNodes.size(); i++) {
```

Comment:
Is this a performance improvement? I think this is unnecessary.

Reply:
This is to avoid potential concurrent modification errors since we may have to define new record type defs while iterating through the current list.